src/compiler.ts

```diff
@@ -361,7 +361,10 @@ export class Compiler extends DiagnosticEmitter {
       this.memoryOffset = i64_new(options.memoryBase);
       module.setLowMemoryUnused(false);
     } else {
-      if (options.optimizeLevelHint >= 2) {
+      if (options.shrinkLevelHint >= 2 || (
+        options.shrinkLevelHint == 0 &&
+        options.optimizeLevelHint >= 3
+      )) {
```

Comment:
What about `-O3s`, shouldn't that also do the optimization? Perhaps `if (shrinkLevel >= 2 || optimizeLevel >= 3)`?

Reply:
I think `-O3s` which is default not necessary for this. I tested some code which affected from this low memory unused pass and speedup not observed, just code size reduction but with slowee compilation speed caveat. So I prefer avoid it for default `--optimize` scenario

Reply:
I see, hmm. Still wondering where such a significant slowdown can come from for something that appears to be such a simple pass. Might not be seeing the full picture, but if my gut feeling is right, there might be something wrong with that pass.

Reply:
From a quick test, it appears that utilizing lowMemoryUnused in the test suite (which uses `-O`) increases run time from 31.5 to 32 seconds on my machine. Tested with `npm run test:compiler -- --parallel --noDiff` on current master, while disabling the pass in the comparison run. That's about 1.6%, which doesn't seem worth to avoid in `-O` if it also reduces size?